{"title": "不同用户python path 权限问题 - 个人文章 ", "index": "python", "content": "问题背景\nroot 用户可以运行pip install mkdocs但gitlab-runner 运行pip install mkdocs 则下列错误\npip install mkdocs\nTraceback (most recent call last):\n  File \"/bin/pip\", line 7, in <module>\n    from pip._internal import main\nImportError: No module named 'pip._internal'\n经过排查，发现是gitlab-runner 在某些python path 的权限问题\n解决方法\n1. cat /bin/pip 查看使用的python 代码\n#!/usr/bin/python3\n\n# -*- coding: utf-8 -*-\nimport re\nimport sys\n\nfrom pip._internal import main\n\nif __name__ == '__main__':\n    sys.argv[0] = re.sub(r'(-script\\.pyw?|\\.exe)?$', '', sys.argv[0])\n    sys.exit(main())\n可知，使用的是python3\n查看path 路径\n切换到root 用户然后在命令行运行：\n[root@test75vm1 ~]# python3\nPython 3.4.5 (default, May 29 2017, 15:17:55) \n[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\n>>> import sys\n>>> sys.path\n['', '/usr/lib64/python34.zip', '/usr/lib64/python3.4', '/usr/lib64/python3.4/plat-linux', '/usr/lib64/python3.4/lib-dynload', '/usr/lib64/python3.4/site-packages', '/usr/lib/python3.4/site-packages']\ncheck path 路径的权限\nls -rlt /usr/lib/python3.4/site-packages\n修改python path 路径的权限\nls -rlt 发现有写目录是没有执行权限的。执行命令\nchmod 755 -R /usr/lib64/python3.4 /usr/lib64/python3.4/plat-linux /usr/lib64/python3.4/lib-dynload /usr/lib64/python3.4/site-packages /usr/lib/python3.4/site-packages\n\n                ", "mainLikeNum": ["0 "], "mainBookmarkNum": "0"}